src/App.tsx

```diff
@@ -14,11 +15,87 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selected: string[],
+};
+
+class App extends Component<{}, State> {
+  state = {
+    selected: ['Jam'],
+  };
+
+  removeGood = (good: string) => {
+    this.setState(state => (
+      { selected: state.selected.filter(elem => elem !== good) }
+    ));
+  };
+
+  addGood = (good: string) => {
+    this.setState(state => (
+      { selected: [...state.selected, good] }
+    ));
+  };
+
```

Comment:
everything is pretty good, but u forgot to add the Clear method and button, check Readme(points 5 and 6)